contracts/samples/TokenPaymaster.sol

```diff
@@ -1,113 +1,212 @@
+    function setUniswapConfiguration(
+        UniswapHelperConfig memory _uniswapHelperConfig
+    ) external onlyOwner {
+        _setUniswapHelperConfiguration(_uniswapHelperConfig);
     }
 
-    //Note: this method assumes a fixed ratio of token-to-eth. subclass should override to supply oracle
-    // or a setter.
-    function getTokenValueOfEth(uint256 valueEth) internal view virtual returns (uint256 valueToken) {
-        return valueEth / 100;
+    /// @notice Allows the contract owner to withdraw a specified amount of tokens from the contract.
+    /// @param to The address to transfer the tokens to.
+    /// @param amount The amount of tokens to transfer.
+    function withdrawToken(address to, uint256 amount) external onlyOwner {
+        SafeERC20.safeTransfer(token, to, amount);
     }
 
-    /**
-      * validate the request:
-      * if this is a constructor call, make sure it is a known account.
-      * verify the sender has enough tokens.
-      * (since the paymaster is also the token, there is no notion of "approval")
-      */
-    function _validatePaymasterUserOp(UserOperation calldata userOp, bytes32 /*userOpHash*/, uint256 requiredPreFund)
-    internal view override returns (bytes memory context, uint256 validationData) {
-        uint256 tokenPrefund = getTokenValueOfEth(requiredPreFund);
-
-        // verificationGasLimit is dual-purposed, as gas limit for postOp. make sure it is high enough
-        // make sure that verificationGasLimit is high enough to handle postOp
-        require(userOp.verificationGasLimit > COST_OF_POST, "TokenPaymaster: gas too low for postOp");
-
-        if (userOp.initCode.length != 0) {
-            _validateConstructor(userOp);
-            require(balanceOf(userOp.sender) >= tokenPrefund, "TokenPaymaster: no balance (pre-create)");
-        } else {
-
-            require(balanceOf(userOp.sender) >= tokenPrefund, "TokenPaymaster: no balance");
+    /// @notice Validates a paymaster user operation and calculates the required token amount for the transaction.
+    /// @param userOp The user operation data.
+    /// @param requiredPreFund The amount of tokens required for pre-funding.
+    /// @return context The context containing the token amount and user sender address (if applicable).
+    /// @return validationResult A uint256 value indicating the result of the validation (always 0 in this implementation).
+    function _validatePaymasterUserOp(UserOperation calldata userOp, bytes32, uint256 requiredPreFund)
+    internal
+    override
+    returns (bytes memory context, uint256 validationResult) {unchecked {
+            uint256 priceMarkup = tokenPaymasterConfig.priceMarkup;
+            uint256 paymasterAndDataLength = userOp.paymasterAndData.length - 20;
+            require(paymasterAndDataLength == 0 || paymasterAndDataLength == 32,
+                "TPM: invalid data length"
+            );
+            uint256 gasPrice = userOp.gasPrice();
```

Comment:
this will revert on the bundler
cc @leekt

Reply:
yup this will revert for sure